src/Widget/ListInput.php

```diff
@@ -32,17 +41,120 @@ final class ListInput extends Widget
     public function run(): string
```

Comment:
Because I pass the `renderSelectOptions(`) method from `HTML` to the `Widget`, that's wrong, it's not the widget's job, it's passing all the `HTML` logic to the widgets.

Reply:
I think this temporary. `ListInput` need refactoring and make him consistent with the `yiisoft/html` package.

Reply:
The problem is not `yiisoft/form`, the problem is `yiisoft/html`, because if you are refactoring `html` you don't fix it instead of transferring the problem to the `widget`.

Reply:
ListInput need split to several class and make receiving incoming data as well as yiisoft/html.